src/helpers.py

```diff
@@ -0,0 +1,132 @@
+import json
+
+from dulwich.diff_tree import TreeChange
+from dulwich.repo import Repo
+from dulwich.porcelain import clone
+from dulwich.walk import WalkEntry
+import difflib
+from tqdm import tqdm
+
+# work with git
+def get_repository_info(url: str):
+    """
+    Returns repository description.
+    Result structure: {
+        'url': <repository url>,
+        'commits': {
+            <commit sha>: {
+                'author': <author name and email>,
+                'changes': [{
+                        'file': <path to changed file>,
+                        'blob_id': <blob id>,
+                        'added': <count of added rows>,
+                        'deleted': <count of deleted rows>
+                    },
+                    ...
+                ]
+            },
+            ...
+        }
+    }
+    :param url: repository url
+    :return: dict with repository description
+    """
+    repo = clone(url)
+
+    res = {
+        'url': url,
+        'commits': {}
+    }
+
+    # walk through changes
+    for entry in tqdm(repo.get_walker()):
+        commit = entry.commit
+        commit_sha = commit.sha().hexdigest()
+        if len(repo.get_parents(commit_sha, commit)) > 1:
+            continue
+        if commit_sha not in res['commits'].keys():
```

Comment:
let's use double quotes everywhere - and add linter to check it https://github.com/zheller/flake8-quotes